### PR TITLE
SAR-5164: Added free filter and updated panel layout as per prototype

### PR DESCRIPTION
**Pull request by brennaksamuel**

### CODE REVIEW

File: app/views/software_choices_filter.scala.html [hunk 1]
```diff
@@ -105,15 +116,14 @@ <h1 class="heading-xlarge">@messages("softwareChoices.filter.title")</h1>
...
                     FilterTemplateModel(
-                        messages("softwareChoices.filter.typeOfSoftware"),
-                        Seq(
-                            FilterFieldModel(form(accounting), messages("softwareChoices.filter.accountingSoftware")),
-                            FilterFieldModel(form(spreadsheets), messages("softwareChoices.filter.spreadSheets"))
+                        messages("softwareChoices.filter.price"),
+                        Seq(FilterFieldModel(form(free), messages("softwareChoices.filter.freeVersion"))
                         ),
-                        appConfig.features.typeOfSoftwareFilterEnabled()
+                    appConfig.features.typeOfSoftwareFilterEnabled()
```
Comment (r-melvin) [inline app/views/software_choices_filter.scala.html]: Should this be its own option?

File: test/views/SoftwareChoicesFilterViewSpec.scala [hunk 2]
```diff
@@ -54,6 +55,8 @@ class SoftwareChoicesFilterViewSpec extends ViewBaseSpec with SoftwareProvidersT
     val accordionSubHeading2 = (i:Int) => s"details > div > h2:nth-child($i)"
     val accordionBullet3 = (i:Int) => s"details > div > ul > li:nth-child($i)"
     val accordionBullet4 = (i:Int) => s"details > div > ul > li:nth-child($i)"
+    val price = """ """
```
Comment (r-melvin) [inline test/views/SoftwareChoicesFilterViewSpec.scala]: Is this needed?

File: test/views/SoftwareChoicesFilterViewSpec.scala [hunk 3]
```diff
@@ -175,10 +178,25 @@ class SoftwareChoicesFilterViewSpec extends ViewBaseSpec with SoftwareProvidersT
...
+        "Filter Header contains Connects to spreadsheets" in {
+          document.select(Selectors.spreadsheetsFilter).text shouldBe FilterSearchMessages.spreadSheets
+        }
+      }
+
       "contain a filter for Additional Features" should {
 
         "Suitable for should contain Additional software features filter header" in {
-          document.select(Selectors.filterHeader(2)).text shouldBe FilterSearchMessages.additionalSoftwareFeatures
+          document.select(Selectors.filterHeader(3)).text shouldBe FilterSearchMessages.additionalSoftwareFeatures
```
Comment (palequezal) [inline test/views/SoftwareChoicesFilterViewSpec.scala]: I know this isn't something you added, but I'd warn against using an index to select the right filter header. They could get out of step and end up not testing the correct thing - there's no guarantee on the left hand side of the assertion statement that we're checking the right one. 

I'd suggest updating the filter helper to add a unique ID to each filter section & then document.select to explicitly grab that ID.

i.e. if the ID was "price-filter", in the test you'd write `document.select("price-filter` h2").text shouldBe ... ` Actually, you could stick the selector in a constant like you have with the rest to be consistent instead, e.g. `document.select(Selectors.priceFilterHeader).text shouldBe ...`

File: app/views/software_choices_filter.scala.html (hunk 1 above)
Comment (palequezal) [reply app/views/software_choices_filter.scala.html]: Agreed, unless there's a business reason why the Price filter should only be shown if the Type of Software filter is displayed, it should have its own toggle in the config

File: test/views/SoftwareChoicesFilterViewSpec.scala [hunk 4]
```diff
@@ -194,25 +212,21 @@ class SoftwareChoicesFilterViewSpec extends ViewBaseSpec with SoftwareProvidersT
...
+          document.select(Selectors.filterHeader(4)).text shouldBe FilterSearchMessages.price
         }
 
-        "Filter Header contains Accounting software" in {
-          document.select(Selectors.accountingFilter).text shouldBe FilterSearchMessages.accountingSoftware
-        }
-
-        "Filter Header contains Connects to spreadsheets" in {
-          document.select(Selectors.spreadsheetsFilter).text shouldBe FilterSearchMessages.spreadSheets
+        "Filter Header contains Free version" in {
```
Comment (palequezal) [inline test/views/SoftwareChoicesFilterViewSpec.scala]: This should refer to the label, not the header

File: test/views/SoftwareChoicesFilterViewSpec.scala [hunk 5]
```diff
@@ -194,25 +212,21 @@ class SoftwareChoicesFilterViewSpec extends ViewBaseSpec with SoftwareProvidersT
         }
       }
 
-      "contain a filter for Type of software" should {
+      "contain a filter for Price" should {
 
-        "Suitable for should contain Type of software filter header" in {
-          document.select(Selectors.filterHeader(3)).text shouldBe FilterSearchMessages.typeOfSoftware
+        "Suitable for should contain Price filter header" in {
```
Comment (palequezal) [inline test/views/SoftwareChoicesFilterViewSpec.scala]: Test description is incorrect

File: test/views/SoftwareChoicesFilterViewSpec.scala [hunk 6]
```diff
@@ -194,25 +212,21 @@ class SoftwareChoicesFilterViewSpec extends ViewBaseSpec with SoftwareProvidersT
...
-        "Filter Header contains Connects to spreadsheets" in {
-          document.select(Selectors.spreadsheetsFilter).text shouldBe FilterSearchMessages.spreadSheets
+        "Filter Header contains Free version" in {
+          document.select(Selectors.freeFilter).text shouldBe FilterSearchMessages.free
         }
       }
 
       "contain a filter for Accessibility features" should {
 
         "Suitable for should contain Accessibility features filter header" in {
```
Comment (palequezal) [inline test/views/SoftwareChoicesFilterViewSpec.scala]: Not one of your changes, but the test description is also incorrect here. They could do with updating so they all refer to the right filter if poss.

**Pull request merged**